fabtests/component/sock_test.c

```diff
@@ -0,0 +1,253 @@
+#include <stdio.h>
+#include <string.h>  
+#include <stdlib.h>
+#include <errno.h>
+#include <unistd.h>   
+#include <arpa/inet.h>    
+#include <sys/types.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <sys/time.h>
+#include <netdb.h>
+#include <time.h>
+#include <poll.h>
+#include <sys/epoll.h>
+
+#define PORT 8888
+
+size_t server_addr_len;
+
+int get_server_addr(char* server_addr, struct sockaddr_storage *server_sock, int *sock)

```

Comment:
All functions in this file should be static.  Please limit line lengths to about 80 characters. Note that this function does more than get the server address.  It's opening the server socket. Move the listen call into here and call this start_server(), which aligns with how other fabtests are structured.